core/domain/classifier_services.py

```diff
@@ -127,6 +127,69 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+
+
+def add_to_training_queue(exploration):
+    """Checks all states of the exploration for re-training conditions and
+    creates ClassifierTrainingJobModel instances in the datastore
+    accordingly.
+
+    Args:
+        exploration: Exploration. Domain object for an exploration.
+    """
+    states = exploration.states
+    for state_name in states:
+        state = states[state_name]
+        if not state.can_undergo_classification():
+            continue
+        exp_id = exploration.id
+        exp_version = exploration.version
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        classifier = get_classifier_from_exploration_attributes(
+            exp_id, exp_version, state_name)
+        if classifier:
```

Comment:
@seanlip . The exp_version that enters this function is the old one, not the new one. If you notice, version gets incremented at the very end of the save_exploration() function and I planned to keep this function just before it. So, we retrieve the old classifier, if it exists.

I think I may not have explained the context properly. This add_to_training_queue function will be placed inside save_exploration() just before the part where model gets updated. This ensures that all checks required for the save_exploration() and version increment are satisfied. So, there is no issue of the version increment failing after this function creates a job.